### PR TITLE
upgrade typescript to 4.4

### LINKED ISSUE #200: TypeScript import from suggestions no longer works after enabling lit-plugin
https://user-images.githubusercontent.com/260340/133578989-1d30d980-3e4d-4b11-943d-d2aac37c6e1a.mov

https://user-images.githubusercontent.com/260340/133578779-1a73a8dd-fa18-4aed-8065-1715c85a403f.mov

### PR DESCRIPTION
**Pull request by 43081j**
Upgrading to typescript 4.4

Fixes #200 

Also upgraded all the eslint dependencies because the guy who maintains es-abstract (depended on by eslint-plugin-import) pushed a funky version latest node doesn't like, the newer one is fine.

Depends on runem/web-component-analyzer#231 as the typescript version has to match what is in web-component-analyzer to build.

It also upgrades ts-simple-type to the `next` tag which uses typescript 4.x.

@justinfagnani ts-simple-type will need publishing before this can be merged, same with web-component-analyzer.

Draft until then

### CODE REVIEW

**Comment by 43081j** [comment]
@rictic  i've rebased this, but we need to publish a new version of web-component-analyzer so it'll have the same typescript version constraint. could you possibly put that on your todo list?

if we get WCA published, we can get this passing and merged, then publish a new lit-analyzer version which fixes the problems everyone currently has

File: packages/lit-analyzer/src/lib/analyze/util/array-util.ts [hunk 1]
```diff
@@ -11,7 +11,11 @@ export function arrayFlat<T>(items: (T[] | T)[]): T[] {
 
 	const flattenArray: T[] = [];
 	for (const item of items) {
-		flattenArray.push(...item);
+		if (Array.isArray(item)) {
+			flattenArray.push(...item);
```
[the remark this answers is not in the document]
Comment (43081j) [reply packages/lit-analyzer/src/lib/analyze/util/array-util.ts:15]: are you sure?

```ts
[[1, 2], [[[[4, 5], 6], 7], 8]].flat();

// results in [1, 2, Array, 8]
```

[the remark this answers is not in the document]
Comment (43081j) [reply packages/lit-analyzer/src/lib/analyze/util/array-util.ts:15]: ah it has a depth parameter apparently! TIL

**Comment by 43081j** [comment]
it builds!

with #240 the bug that is blocking people from using this should be fixed at least when we merge.

it'd be good if we could have a test for the actual problem but i wasn't sure the best way to do it. as in call the extension with the extra param and ensure it makes it through to the previous language service.

**Comment by rictic** [comment]
Yeah, I got a ways into setting up an integration test of that case but it's surprisingly tricky. The vscode API for extension testing does not seem to make it easy to load two extensions at once to test potential conflicts

**Review by rictic — APPROVED** [review]

**Comment by rictic** [comment]
Nice change, and thank you for all of your great work on updating all of these deps!!

**Comment by christophe-g** [comment]
Thank you !

**Pull request merged**